src/components/ChakraThemeProvider.tsx

```diff
@@ -0,0 +1,34 @@
+import { PropsWithChildren, useMemo } from "react"
+import merge from "lodash/merge"
+import { useTheme as useNextTheme } from "next-themes"
+import { ColorMode, ColorModeContext, ThemeProvider } from "@chakra-ui/react"
+
+import customTheme from "@/@chakra-ui/theme"
+
+import { useLocaleDirection } from "@/hooks/useLocaleDirection"
+
+function ChakraThemeProvider({ children }: PropsWithChildren) {
+  // get the theme from next-themes
+  const { resolvedTheme } = useNextTheme()
+
+  const direction = useLocaleDirection()
+
+  const theme = useMemo(() => merge(customTheme, { direction }), [direction])
+
+  return (
+    <ThemeProvider theme={theme}>
```

Comment:
Hmm good points.

Lets see, the main issue I have with `ChakraBaseProvider` is that is [declaring the `ColorModeProvider`](https://github.com/chakra-ui/chakra-ui/blob/v2/packages/components/src/provider/provider.tsx#L91) and is not on our control. But i see your point....have you seen issues with the current implementation? I mean in terms of styles. Maybe I haven't seen any issue atm because we haven't migrated anything yet to Shadcn 🤔 

> You will need to update how dark mode styling is applied. Instead of targeting a class attribute, target data-theme. This would be declared in the attribute prop of the theme provider from next-themes and replacing .dark selector in main.css with [data-theme="dark"].

I couldn't follow you 100% here. Open to change it to use the data- attr, didn't understand too much why, but not a big deal. Can do that if that aligns with your setup.

Reply:
I'll clarify.

Yes, I see where flipping to `ChakraBaseProvider` was breaking the sync for color mode. Chakra v2 does compiling in such a way where it looks at `data-theme` to update styling for color mode. 

`.chakra-ui-dark` is a targeted class for dark mode, but only if there is not a `data-theme` rendered on `html`. And, `.chakra-ui-dark` is at the `body` tag. This means that without `data-theme` the chakra color styling props set on the `body` will not render.

So it's the following combination:
- `NextThemesProvider` should receive `attribute="data-theme"`
- `ChakraBaseProvider` to be used instead of `ThemeProvider` from emotion to ensure correct theme compiling.
- `.dark` -> `[data-theme="dark"]` in `main.css` because that is the selector being targeted.

I think this would be a cleaner way to go, so styles are not moved around for the sake of accommodation, and possibly forgotten about later.

Reply:
Uhmm okok, I think I got it now. Sounds good and cleaner. I also think I could also remove the `ColorModeScript`.

Reply:
@pettinarip

Ok!

Forgot to mention that the Tailwind config would also need to be updated to reflect this.

Reply:
I've updated the PR with these changes and so far, most of the things are working as expected (+ simpler code), except the places where we are use `useColorModeValue` or `useColorMode`.

I think this is because Chakra's internal context is not updating the state for the selected theme since we are no longer controlling the value and we are switching the theme using the `setTheme` of `next-themes`.

Ideally, I would like to somehow sync this state with the next-themes. LMK if you have something in mind.

If not, we could:
1. just replace `useColorModeValue` or `useColorMode` with the a custom function using `useTheme` (would lean towards this option if we can't control the state)
2. or we could sync manually the themes by setting the theme in `next-themes` and `chakra` when we toggle (worst option imo)

Reply:
I think it's weird that it's losing context here. I would be thinking custom function here. It should be pretty straightforward.

See [see source code for these hooks](https://github.com/chakra-ui/chakra-ui/blob/v2/packages/components/src/color-mode/color-mode-context.ts).

Reply:
Yes, exactly. I think I will create one and replace all these instances. Sooner or later we will have to do the same thing anyway.